controllers/admin/AdminCustomerThreadsController.php

```diff
@@ -1017,7 +1017,7 @@ public function syncImap()
         $str_errors = '';
         $str_error_delete = '';
 
-        if (count($errors) && is_array($errors)) {
+        if (!empty($errors)) {
```

Comment:
```suggestion
        if (is_array($errors) && count($errors)) {
```

Should do the job. With your change, you don't check if $errors is an array and is iterable.

Reply:
I don't think it's that needed for a `imap_errors(): array|false`, but I'll follow your advice.

Reply:
@PierreRambaud Are you really sure about this? I think that the `$errors` variable type is forced to be an array OR false as it's an `imap_errors`, so empty is the perfect case, if $errors is false, empty is false, if it's an array but empty it will return false, looks like `is_array($errors) && count($errors)` is overkill (except if it has better performances than using empty)

ping @Progi1984 looks like the thing we seen together last week

Reply:
The first `is_array` check will be faster than `empty` meaning it will fail fast
I didn't check it came from the `imap_errors`, so both are valid ^^